tests/unit/test_utils.py

```diff
@@ -154,3 +157,21 @@ def test_datetime_to_utc_as_str(target: str, is_pytz: bool):
 
     res = datetime_to_utc_as_str(target)
     assert res == EXPECTED_DATETIME
+
+
+@mock.patch("latigo.utils.ThreadPoolExecutor")
+def test_get_thread_pool_executor(mocked_thread_pool_executor):
+    executor = get_thread_pool_executor(2)
+    cached_executor = get_thread_pool_executor(2)
+
+    assert executor == cached_executor
```

Comment:
```suggestion
    assert executor is cached_executor
```
This might make a little bit more sense.

Reply:
thanks, adjusted